### PR TITLE
feature/issue 1752

### LINKED ISSUE #1752: Add code example snippet to cogent3.app.sample.omit_duplicated
The code should be executable by doctesting. See `omit_degenerates()`, `omit_gap_pos()`, `concat()` docstrings for examples. The purpose of these examples is a user gets help on an app and the help displays working code. In other words, the sample code you write should work in an interactive interpreter. It should also not load any data from file, but create sequence collections (in this case) from dictionaries with small sequences.

This app removes redundant sequences from a sequence collection or alignment, i.e. if `seq1` is "ACGG" and `seq2` is also "ACGG", then one of those would be removed from the collection and the dropped sequence name added to the `<collection>.info` object.

See the `test_omit_duplicated_aligned` method in `tests/test_app/test_sample.py` for examples. You should only use a minimal set of data to demonstrate how it works, not the entire data set used for testing.

### PR DESCRIPTION
**Pull request by Ruizhe-wang**

### CODE REVIEW

**Comment by Ruizhe-wang** [comment]
DOC: add code example snippet to cogent3.app.sample.omit_duplicated, fixes #1752

**Comment by Ruizhe-wang** [comment]
DOC: add code example snippet to cogent3.app.sample.omit_duplicated, fixes https://github.com/cogent3/cogent3/issues/1752

File: src/cogent3/app/sample.py [hunk 1]
```diff
@@ -754,6 +754,58 @@ def __init__(self, mask_degen=False, choose="longest", seed=None, moltype=None):
...
+        >>> app = get_app("omit_duplicated", moltype="dna", choose="longest")
+        >>> seqs = make_aligned_seqs(seq_collection, moltype="DNA")
+        >>> result = app(seqs)
+        >>> print(result.to_pretty())
+        a    ACGT
+        b    ...-
+        d    ...G
+        e    .GTC    
+
+        Exclude all members of duplicated sets:
```
Comment (fredjaya) [inline src/cogent3/app/sample.py:785]: Replace with 

```
Create an app to exclude all duplicate sequences from the collection.
```

File: src/cogent3/app/sample.py [hunk 2]
```diff
@@ -754,6 +754,58 @@ def __init__(self, mask_degen=False, choose="longest", seed=None, moltype=None):
...
+        >>> print(result.to_pretty())
+        a    ACGT
+        b    ...-
+        e    .GTC
+
+
+        Ignore degenerate characters when identifying duplicates:
+
+        >>> aln_dna = make_aligned_seqs({
+        "seq1": "ATCG",
```
Comment (fredjaya) [inline src/cogent3/app/sample.py:798]: Add leading `...`

**Comment by fredjaya** [comment]
Thanks for the contribution!

I have suggested a few improvements for the example descriptions. 

After adding docstring examples, check that they work with `doctest`. A few errors came up regarding the multi-line code missing `...` at the beginning of the lines. Instructions for testing can be found [here](https://github.com/cogent3/c3dev/wiki/How%E2%80%90to-write-docstring-examples-for-methods-and-functions#how-to-test-docstring-examples).

Let me know here if you have any questions at all!

**Review by fredjaya — CHANGES_REQUESTED** [review]
Thanks for the contribution!

I have suggested a few improvements for the example descriptions. 

After adding docstring examples, check that they work with `doctest`. A few errors came up regarding the multi-line code missing `...` at the beginning of the lines. Instructions for testing can be found [here](https://github.com/cogent3/c3dev/wiki/How%E2%80%90to-write-docstring-examples-for-methods-and-functions#how-to-test-docstring-examples).

Let me know here if you have any questions at all!

**Comment by Ruizhe-wang** [comment]
Hi,
Thank you for your guidance!
I have completed the improvements you suggested. Please let me know if there are any further modifications needed. And I also notice when I commit in Github desktop, an email from Github "CI: Some jobs were not successful", should I worry about it? @fredjaya

File: src/cogent3/app/sample.py [hunk 3]
```diff
@@ -754,6 +754,61 @@ def __init__(self, mask_degen=False, choose="longest", seed=None, moltype=None):
...
+        >>> print(result.to_pretty())
+        a    ACGT
+        b    ...-
+        e    .GTC
+
+
+        Create sample data with degenerate characters, and an app to ignore
+        these when identifying duplicates.
+
+        >>> aln_dna = make_aligned_seqs({
```
Comment (fredjaya) [inline src/cogent3/app/sample.py:800]: Replace with

```
>>> aln_dna = make_aligned_seqs({
... "seq1": "ATCG", 
... "seq2": "ATYG", 
... "seq3": "GGTA",
... "seq4": "GGTA"
... }, 
... moltype="DNA")
```

**Review by fredjaya — CHANGES_REQUESTED** [review]
Hi Ruizhe, thanks for the updates, we are nearly there!

`doctest` threw errors with the full stops used, where `·` is a different symbol to `.` 

```python
ord(".") # 46, correct one
ord(".") # 183
```

I have provided snippets for you to copy and paste throughout.

**Comment by fredjaya** [comment]
The GitHub actions haven't been run so it looks like an error, no need to worry.

**Pull request merged**